fastly/resource_fastly_alert.go

```diff
@@ -291,11 +296,25 @@ func buildDimensions(data map[string][]string, v map[string]any) map[string][]st
 }
 
 func buildEvaluationStrategy(v map[string]any) map[string]any {
-	return map[string]any{
-		"type":      v["type"].(string),
-		"period":    v["period"].(string),
-		"threshold": v["threshold"].(float64),
+	evaluationStrategy := map[string]any{}
+
+	if value, ok := v["type"]; ok {
+		evaluationStrategy["type"] = value.(string)
+	}
+
+	if value, ok := v["period"]; ok {
+		evaluationStrategy["period"] = value.(string)
 	}
+
+	if value, ok := v["threshold"]; ok {
+		evaluationStrategy["threshold"] = value.(float64)
+	}
```

Comment:
The key lookups in the map are redundant here as we specify those fields as `required` in the schema.

So I'd rework this like so:

```go
// Required attributes
m := map[string]any{
	"type":      v["type"].(string),
	"period":    v["period"].(string),
	"threshold": v["threshold"].(float64),
}
// Optional attributes
if value, ok := v["ignore_below"]; ok {
	evaluationStrategy["ignore_below"] = value.(float64)
}
return m
```